netmiko/nokia/nokia_sros_ssh.py

```diff
@@ -168,3 +170,72 @@ def cleanup(self, command="logout"):
         # Always try to send final 'logout'.
         self._session_log_fin = True
         self.write_channel(command + self.RETURN)
+
+
+class NokiaSrosFileTransfer(BaseFileTransfer):
+    def remote_space_available(self, search_pattern=r"(\d+) \w+ free"):
+        """Return space available on remote device."""
+
+        remote_cmd = "file dir {}".format(self.file_system)
+        remote_output = self.ssh_ctl_chan.send_command_expect(remote_cmd)
+        match = re.search(search_pattern, remote_output)
+        return int(match.group(1))
+
+    def check_file_exists(self, remote_cmd=""):
+        """Check if destination file exists (returns boolean)."""
+
+        if self.direction == "put":
+            remote_cmd = "file dir {}/{}".format(self.file_system, self.dest_file)
+            remote_out = self.ssh_ctl_chan.send_command_expect(remote_cmd)
+            if "File Not Found" in remote_out:
+                return False
+            elif self.dest_file in remote_out:
+                return True
+            else:
+                raise ValueError("Unexpected output from check_file_exists")
+        elif self.direction == "get":
+            return os.path.exists(self.dest_file)
+
+    def remote_file_size(self, remote_cmd=None, remote_file=None):
+        """Get the file size of the remote file."""
+
+        if remote_file is None:
+            if self.direction == "put":
+                remote_file = self.dest_file
+            elif self.direction == "get":
+                remote_file = self.source_file
+        if not remote_cmd:
+            remote_cmd = "file dir {}/{}".format(self.file_system, remote_file)
+        remote_out = self.ssh_ctl_chan.send_command(remote_cmd)
+
+        if "File Not Found" in remote_out:
+            raise IOError("Unable to find file on remote system")
+
+        # Parse dir output for filename. Output format is:
+        # "10/16/2019  10:00p                6738 {filename}"
+
+        pattern = r"(\S+)[ \t]+(\S+)[ \t]+(\d+)[ \t]+{}".format(re.escape(remote_file))
+        match = re.search(pattern, remote_out)
+
+        if not match:
+            raise ValueError("Filename entry not found in dir output")
+
+        file_size = int(match.group(3))
+        return file_size
+
+    def verify_file(self):
+        """Verify the file has been transferred correctly based on filesize."""
+        if self.direction == "put":
+            return os.stat(self.source_file).st_size == self.remote_file_size(
+                remote_file=self.source_file
+            )
+        elif self.direction == "get":
+            return (
+                self.remote_file_size(remote_file=self.source_file)
+                == os.stat(self.source_file).st_size
+            )
```

Comment:
So SR-OS doesn't support calculating any Hash on the files (MD5/SHA1)?

Reply:
No, there is no support for md5 hash calculation.